### PR TITLE
docs(php): update documentation about extensions and dependencies

### PR DESCRIPTION
**Pull request by Frzk**
This PR does a few things:
- move the documentation related to PHP extensions to its own page.
- move the documentation related to Composer and dependencies to its own page.
- update the documentation related to PHP extensions.
- update the documentation related to Composer and dependencies.
- update the start page that used to contain these documentations.
- rework the pages index to have something more consistent.

### CODE REVIEW

File: src/_posts/languages/php/2000-01-01-dependencies.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+- `2.3.10`
+- `2.4.4`
+- `2.5.7`
+
+## Working with Composer Environments
+
+By default, Scalingo considers that your application runs in *production* mode.
+This means that `composer install` automatically runs with the `--no-dev` flag,
+and, as a result, won't install the development dependencies of your
+application, if any.
```
Comment (EtienneM) [inline src/_posts/languages/php/2000-01-01-dependencies.md:102]: suggestion(non-blocking): this sentence is really long, hence hard to read. Maybe split it into multiple, smaller, sentences?

```suggestion
This means that `composer install` automatically runs with the `--no-dev` flag.
As a result, Composer won't install the development dependencies of your
application, if any.
```

File: src/_posts/languages/php/2000-01-01-dependencies.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+If, for some reason, you would like to run your application with these
+development dependencies installed (e.g. to debug your app), please set the
+`COMPOSER_DEV` environment variable to `true`.
```
Comment (EtienneM) [inline src/_posts/languages/php/2000-01-01-dependencies.md:106]: suggestion(non-blocking): "for some reason" seems to only make the sentence harder to read without adding useful information. I would remove it.
```suggestion
Set the `COMPOSER_DEV` environment variable to `true` if you would like to run your application with these
development dependencies installed (e.g. to debug your app).
```

File: src/_posts/languages/php/2000-01-01-extensions.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  "require": {
+    "ext-mongodb": "*",
+    "ext-imagick": "*",
+    ...
+  }
+}
+```
+
+Once done, don't forget to generate the `composer.lock` file and to commit it
+to your codebase.
```
Comment (EtienneM) [inline src/_posts/languages/php/2000-01-01-extensions.md:60]: question: I don't think adding an extension changes the `composer.lock`. I think it's only an instruction for the platform.

File: src/_posts/languages/php/2000-01-01-extensions.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+```bash
+composer install
+```
+
+As recommended, we will use the multi-buildpack to tell the platform to:
+
+1. first, call the apt-buildpack to install the required packages;
+2. and then, use the php-buildpack to install PHP, Composer and the required
+   extensions.
```
Comment (EtienneM) [inline src/_posts/languages/php/2000-01-01-extensions.md:124]: wording: this is how we refer to the buildpacks in the doc

```suggestion
1. first, call the APT buildpack to install the required packages;
2. and then, use the PHP buildpack to install PHP, Composer and the required
   extensions.
```

**Review by EtienneM — COMMENTED** [review]
> I could maybe also fix https://github.com/Scalingo/php-buildpack/issues/310 in this PR? :wondering:

There is a documentation ticket if you want to assign it to yourself (#2028) :)

File: src/_posts/languages/php/2000-01-01-dependencies.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+In some circumstances, it can be convenient to add either the
+`--ignore-platform-req=` or the `--ignore-platform-reqs` flag to the above
+commands.
+
+After each command, the `composer.lock` file is automatically updated. Don't
+forget to commit the modifications!
+
+{% note %}
+    Both the `scalingo.json` and `scalingo.lock` files are _required_ for the
```
[the remark this answers is not in the document]
Comment (Frzk) [reply src/_posts/languages/php/2000-01-01-dependencies.md]: :fearful:

File: src/_posts/languages/php/2000-01-01-dependencies.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
+---
+title: Managing Dependencies
+nav: Managing Dependencies
+modified_at: 2023-06-02 16:00:00
+tags: php
+index: 4
```
[the remark this answers is not in the document]
Comment (Frzk) [reply src/_posts/languages/php/2000-01-01-dependencies.md]: Alright, done in https://github.com/Scalingo/documentation/pull/2139/commits/829c67bb4f86ff17f20dea99282b1cacf766833f

File: src/_posts/languages/php/2000-01-01-dependencies.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Frzk) [reply src/_posts/languages/php/2000-01-01-dependencies.md]: Fixed in https://github.com/Scalingo/documentation/pull/2139/commits/829c67bb4f86ff17f20dea99282b1cacf766833f - Sorry :(

File: src/_posts/languages/php/2000-01-01-dependencies.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    Both the `scalingo.json` and `scalingo.lock` files are _required_ for the
+    platform to handle your dependencies.
+{% endnote %}
+
+### Managing Private Dependencies
+
+If you want to install a private dependency, you need to define the
+`COMPOSER_AUTH` environment variable, as specified in
+[the Composer documentation](https://getcomposer.org/doc/03-cli.md#composer-auth).
+For instance, for a private dependency hosted on GitHub, the `COMPOSER_AUTH`
```
[the remark this answers is not in the document]
Comment (Frzk) [reply src/_posts/languages/php/2000-01-01-dependencies.md]: Removed in https://github.com/Scalingo/documentation/pull/2139/commits/829c67bb4f86ff17f20dea99282b1cacf766833f

File: src/_posts/languages/php/2000-01-01-dependencies.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+```json
+{
+  "github-oauth": {
+    "github.com": "MY-TOKEN"
+  }
+}
+```
+
+`MY-TOKEN` must be replaced with a valid token of your GitHub account.
```
[the remark this answers is not in the document]
Comment (Frzk) [reply src/_posts/languages/php/2000-01-01-dependencies.md]: I added a small paragraph precising it's an "access token (OAuth token)", I added a link to the GitHub's documentation and I added a link to the WebUI from where one can manage their tokens.
All in https://github.com/Scalingo/documentation/pull/2139/commits/829c67bb4f86ff17f20dea99282b1cacf766833f

File: src/_posts/languages/php/2000-01-01-dependencies.md (hunk 1 above)
Comment (Frzk) [reply src/_posts/languages/php/2000-01-01-dependencies.md]: You're right, applied in https://github.com/Scalingo/documentation/pull/2139/commits/829c67bb4f86ff17f20dea99282b1cacf766833f

File: src/_posts/languages/php/2000-01-01-extensions.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
+---
+title: Managing PHP Extensions
+nav: Managing PHP Extensions
+modified_at: 2023-06-02 16:00:00
+tags: php
+index: 3
```
[the remark this answers is not in the document]
Comment (Frzk) [reply src/_posts/languages/php/2000-01-01-extensions.md]: Done in https://github.com/Scalingo/documentation/pull/2139/commits/829c67bb4f86ff17f20dea99282b1cacf766833f

File: src/_posts/languages/php/2000-01-01-extensions.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+environment variable.
+
+### Full Example
+
+In this example, we are going to install the PECL extension called `yaml`. This
+is a good example as it requires the`libyaml` and `libyaml-dev` packages to be
+installed, and it also needs some specific compilation flags.
+
+First, let's tell the platform we need this extension. To do so, we will have
+to create a file called `composer.json` at the root of our project, like this:
```
[the remark this answers is not in the document]
Comment (Frzk) [reply src/_posts/languages/php/2000-01-01-extensions.md]: Applied everywhere ^^

File: src/_posts/languages/php/2000-01-01-extensions.md [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+```
+
+As recommended, we will use the multi-buildpack to tell the platform to:
+
+1. first, call the apt-buildpack to install the required packages;
+2. and then, use the php-buildpack to install PHP, Composer and the required
+   extensions.
+
+Let's create a file called `.buildpacks` at the root of the project with the
+following content:
```
[the remark this answers is not in the document]
Comment (Frzk) [reply src/_posts/languages/php/2000-01-01-extensions.md]: Same as above, this has been fixed in https://github.com/Scalingo/documentation/pull/2139/commits/829c67bb4f86ff17f20dea99282b1cacf766833f

File: src/_posts/languages/php/2000-01-01-extensions.md [hunk 12]
```diff
@@ -0,0 +1,145 @@
...
+    "ext-yaml": "*"
+  }
+}
+```
+
+Don't forget to generate the `composer.lock` file, using Composer:
+
+```bash
+composer install
+```
```
Comment (EtienneM) [inline src/_posts/languages/php/2000-01-01-extensions.md:115]: Related to previous review: https://github.com/Scalingo/documentation/pull/2139#discussion_r1217750404

```suggestion
```

**Review by EtienneM — APPROVED** [review]
One last comment but already approving this PR. 

You have conflict in the PHP start page.

**Pull request merged**